e2e/wdio/platform/pages/date-picker.po.ts

```diff
@@ -0,0 +1,50 @@
+import { BaseComponentPo } from './base-component.po';
+import { waitForElDisplayed, waitForPresent } from '../../driver/wdio';
+
+export class DatePicker extends BaseComponentPo {
+    url = '/date-picker';
+    root = '#page-content';
+
+    inputDatePicker = 'fd-date-picker input';
+    buttonDatePicker = 'fd-date-picker button';
+
+    activeButtonDatePicker = '[ng-reflect-is-disabled="false"] + span button';
+    activeInputDatePicker = '[ng-reflect-is-disabled="false"][aria-label="Date input"]';
+
+    calendarExpanded = '.fd-popover__popper fd-calendar';
+    calendarYearsSection = '.fd-calendar__content--years';
+
+    currentYear = '[aria-label="Select year"]';
+    currentDay = '//*[contains(@class, "fd-calendar__item--current") or contains(@class, "is-active")]';
+
+    buttonGerman = 'button[ng-reflect-label="German"]'
+    buttonBulgarian = 'button[ng-reflect-label="Bulgarian"]';
+    buttonSelectYear = '[aria-label="Select year"]';
+    buttonSelectMonth = '[aria-label="Select month"]';
+
+    buttonFirstRangeYear = '(//td[contains(@id,"fd-aggregated-year")]/child::span)[1]'
+    buttonFirstYear = '(//td[contains(@id,"year")]/child::span)[1]'
+    buttonFirstMonth = '(//td[contains(@id,"month")]/child::span)[1]'
+
+    filterCaledarValue = (name: string) => {
+        return `//td[contains(@id,"fd-${name}")]`;
+    };
+
+    datePickerInputByIndex = (index: string) => {
```

Comment:
look like it is auto-generated  id's
maybe better use fd-popover tag with index?